doc/azure_ci.md

```diff
@@ -0,0 +1,160 @@
+  - [azure-linux-template.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/.azure/azure-linux-template.yml) : Template for Azure Linux build
+  - [azure-osx-template.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/.azure/azure-osx-template.yml) : Template for Azure macOS build
+  - [azure-windows-template.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/.azure/azure-windows-template.yml) : Template for Azure Windows build
+
+  __NOTE__: It is likely going to be necessary to modify this configuration to
+  match the needs of your project, such as tweaking the build matrix to alter
+  which GHC versions you test against, or to specify GHC-version-specific
+  `stack.yaml` files if necessary. Don't be surprised if it doesn't work the
+  first time around. See the multiple GHC section below for more information.
+
+Each of these configurations is ready to be used immediately. Steps to make them work:
+* For simple Azure configuration, copy-paste the
+  [azure-simple](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/.azure/azure-simple.yml)
+  file into `azure-pipelines.yml` in the root of your repository.
+* For complex Azure configuration, you need to take the above four
+files and put them in the root of your repository.
+
+Once you have done that, you need to create a pipeline from the Azure
+Web interface to get things started.
+
+The rest of this document explains the details of common Azure
+configurations for those of you who want to tweak the above
+configuration files or write your own.
+
+*Note:* both Azure and Stack infrastructures are actively developed. We try to
+ document best practices at the moment.
+
+## Infrastructure
+
+Note that you need atleast one agent to build your code. You can
+specify which virtual image you want to choose using this configuration:
+
+``` yaml
+pool:
+  vmImage: ubuntu-16.04
+```
+
+The other popular options are `macOS-10.13`, `vs2017-win2016` for Mac
+and Windows respectively. You can find the [complete
+list](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=vsts&tabs=yaml)
+here.
+
+Note that as of now, Azure CI doesn't offer any caching support. You
+can use something like [cache-s3](https://github.com/fpco/cache-s3)
+for explicit caching, although it may cost you. For a project with an
+example usage of `cache-s3`, you can see the Azure configuration of
+[stack](https://github.com/commercialhaskell/stack) repository.
+
+## Installing Stack
+
+Currently there is only one reasonable way to install Stack: fetch precompiled
+binary from the Github.
+
+```yaml
+- script: |
+    mkdir -p ~/.local/bin
+    curl -L https://get.haskellstack.org/stable/linux-x86_64.tar.gz | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
```

Comment:
Our best practice recommendation is to specify an exact `stack` version for CI configurations so that builds are reproducible, so maybe the example should do it that way too (although I notice the Travis docs don't, so it's worth discussing).

Reply:
After speaking with Michael, we decided that the current way (latest stack version) is better for the docs.